source/Controlled.tsx

```diff
@@ -627,70 +658,56 @@ class ControlledBase extends React.Component<ControlledPropsWithDefaults, Contro
       window.getComputedStyle(this.imgEl).display !== 'none'
   }
 
+  // ===========================================================================
+
   /**
    * Perform zooming actions
    */
   zoom = () => {
     this.bodyScrollDisable()
     this.refDialog.current?.showModal?.()
+    this.refModalImg.current?.addEventListener?.('transitionend', this.handleImgTransitionEnd) // must be added after showModal
     this.setState({ modalState: ModalState.LOADING })
-    this.loadZoomImg()
-
-    window.addEventListener('touchstart', this.handleTouchStart, { passive: true })
-    window.addEventListener('touchmove', this.handleTouchMove, { passive: true })
-    window.addEventListener('touchend', this.handleTouchEnd, { passive: true })
-    window.addEventListener('touchcancel', this.handleTouchCancel, { passive: true })
-    document.addEventListener('keydown', this.handleKeyDown, true)
-
-    this.refModalImg.current?.addEventListener?.('transitionend', this.handleZoomEnd, { once: true })
   }
 
   /**
-   * Report that the zoom modal is loaded and listen
-   * for window resizing
+   * Perform unzooming actions
    */
-  handleZoomEnd = () => {
-    setTimeout(() => {
-      this.setState({ modalState: ModalState.LOADED })
-      window.addEventListener('resize', this.handleResize, { passive: true })
-      window.addEventListener('wheel', this.handleWheel, { passive: true })
-    }, 0)
+  unzoom = () => {
+    this.setState({ modalState: ModalState.UNLOADING })
   }
 
   // ===========================================================================
 
   /**
-   * Perform unzooming actions
+   * Handle img zoom/unzoom transitionend events and update states:
+   *   - LOADING -> LOADED
+   *   - UNLOADING -> UNLOADED
    */
-  unzoom = () => {
-    this.setState({ modalState: ModalState.UNLOADING })
-
-    window.removeEventListener('wheel', this.handleWheel)
-    window.removeEventListener('touchstart', this.handleTouchStart)
-    window.removeEventListener('touchmove', this.handleTouchMove)
-    window.removeEventListener('touchend', this.handleTouchEnd)
-    window.removeEventListener('touchcancel', this.handleTouchCancel)
-    document.removeEventListener('keydown', this.handleKeyDown, true)
+  handleImgTransitionEnd = () => {
+    clearTimeout(this.timeoutTransitionEnd)
```

Comment:
However we get to this point, if `this.timeoutTransitionEnd` was actually triggered by the `transitionend` event, this makes sure we clear the timeout backstop that ensures it gets called (so we don't call it twice).